versioned/storage/common/src/main/java/org/projectnessie/versioned/storage/common/persist/Persist.java

```diff
@@ -0,0 +1,341 @@
+
+  /**
+   * Like {@link #fetchObj(ObjId)}, but finds multiple objects by name at once, leveraging bulk
+   * queries against databases.
+   *
+   * <p>The behavior when providing the same {@link ObjId} multiple times in {@code ids} is
+   * undefined, implementations may return the same object or equal objects. In any case, providing
+   * the same ID multiple times is discouraged.
+   *
+   * @return The returned objects will be a concrete types according to their {@link ObjType},
+   *     (de)serialization is handled by the database specific implementation of {@link Persist}.
+   *     Returns {@code null} for {@code id} equal to {@link ObjId#EMPTY_OBJ_ID}
+   * @throws ObjNotFoundException with the IDs for which no {@link Obj objects} exist
+   * @see #fetchObjType(ObjId)
+   * @see #fetchTypedObj(ObjId, ObjType, Class)
+   * @see #fetchObj(ObjId)
+   */
+  @Nonnull
+  @jakarta.annotation.Nonnull
+  Obj[] fetchObjs(@Nonnull @jakarta.annotation.Nonnull ObjId[] ids) throws ObjNotFoundException;
+
+  /**
+   * Stores the given object as a new record.
+   *
+   * <p>This is a very low level persist operation. Prefer the persist operations on the various
+   * logic interfaces to ensure a stable and deterministic ID generation.
+   *
+   * @param obj the object to store
+   * @return the non-null object ID if the object was stored as a new record or {@code null} if an
+   *     object with the same ID already exists.
+   * @throws ObjTooLargeException thrown when a hard database row/item size limit has been hit, or a
+   *     "soft" size restriction in {@link #config()}
+   * @see #storeObjs(Obj[])
+   */
+  @Nullable
+  @jakarta.annotation.Nullable
+  default ObjId storeObj(@Nonnull @jakarta.annotation.Nonnull Obj obj) throws ObjTooLargeException {
+    return storeObj(obj, false);
+  }
+
+  /**
+   * Stores the given object as a new record, variant of {@link #storeObj(Obj)} that explicitly
+   * allows ignoring soft object/attribute size restrictions.
+   *
+   * <p>This is a very low level persist operation. Prefer the persist operations on the various
+   * logic interfaces to ensure a stable and deterministic ID generation.
+   *
+   * @param obj the object to store
+   * @param ignoreSoftSizeRestrictions whether to explicitly ignore soft size restrictions, use
+   *     {@code false}, if in doubt
+   * @return the non-null object ID if the object was stored as a new record or {@code null} if an
+   *     object with the same ID already exists.
+   * @throws ObjTooLargeException thrown when a hard database row/item size limit has been hit, or,
+   *     if {@code ignoreSoftSizeRestrictions} is {@code false}, a "soft" size restriction in {@link
+   *     #config()}
+   * @see #storeObjs(Obj[])
+   */
+  @Nullable
+  @jakarta.annotation.Nullable
+  ObjId storeObj(@Nonnull @jakarta.annotation.Nonnull Obj obj, boolean ignoreSoftSizeRestrictions)
```

Comment:
Most (all?) implementations return the `obj.id()` from this method on success and return `null` on exceptions. Replacing an already available ID with `null` to indicate errors looks odd. Would it make sense to return a `boolean` instead?

Reply:
refactored